server/src/main/java/org/opensearch/identity/scopes/ActionScope.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.identity.scopes;
+
+import org.opensearch.identity.scopes.ScopeEnums.ScopeNamespace;
+import org.opensearch.identity.scopes.ScopeEnums.ScopeArea;
+
+/**
+ * This enum defines the ActionScope Scope implementation.
+ *
+ * An ActionScope is composed of a ScopeNamespace, ScopeArea, and Action.
+ * Together, these three components reference a Scope which confers execution privileges to a Subject. For example, the READ ActionScope
+ * is has a ScopeArea of INDEX and an Action of READ. Its ScopeNamespace is ACTION since it is an ActionScope. This ActionScope is required by all
+ * Subjects which seek to perform read-related operations on an INDEX.
+ *
+ * ActionScopes.ALL: Grants access to execute any REST or Transport action in OpenSearch
+ * ActionScopes.READ: Grants access to execute all REST and Transport actions focusing on "read" operations. Specifically GetAction and MultiGetAction
+ *
+ * @opensearch.experimental
+ */
+public enum ActionScope implements Scope {
+    ALL(ScopeArea.CLUSTER, "ALL"),
+    READ(ScopeArea.INDEX, "READ");
```

Comment:
Sorry, I am not sure I understand your suggestion. Could you provide an example of what you were looking for?

Reply:
I mean try to wrap actions into enumeration, may be we should just implement `ActionScope` as class and not enumeration?  For example, `GetAction`'s name is `indices:data/read/get`, we could represent that as (pseudo code):

```
class ActionScope("indices" /* that maps to ScopeArea */,  "data/read/get")
```

Reply:
Oh gotcha. The Enums are used because it is not meant to accept arbitrary input. The idea is that there is a set number of options to select from and anything outside of those options should be ignored. If we swap to a class based version, I believe we will need to enforce matching based off the predefined options anyways in order to maintain this. 

Does that make sense?

Reply:
> I believe we will need to enforce matching based off the predefined options anyways in order to maintain this.

Correct, but those are predefined as XxxAction.NAME, it just looks to me we will end up by creating enum member  for every action

Reply:
Hm .... just to clarify the intent (it makes sense what you are saying in general): right now we could have fine grained control over index's read actions, for example we could let read the index (GetAction) but forbid read index template (GetIndexTemplateAction), as the scope these two fall under "index" : "read" so if we model them as such - we won't be able to differentiate these two read operation.

Reply:
Yeah exactly. The scopes are meant as very broad access controls native to core OpenSearch. For FGAC we would expect someone to use an Identity/Security Plugin.

Reply:
Got it, all clear than, thank you